servant-server/test/Servant/Server/Internal/RoutingApplicationSpec.hs

```diff
@@ -59,3 +100,10 @@ spec = do
       _ <- simpleRun $ delayed (liftIO $ throwIO DivideByZero) (Route $ return ())
       cleanUpDone <- isJust <$> readIORef delayedTestRef
       cleanUpDone `shouldBe` False
+  describe "ResApi" $
+    with (return $ serve resApi resServer) $ do
+      it "writes and cleanups resources" $ do
+        request "GET" "foobar" [] "" `shouldRespondWith` "foobar"
+        liftIO $ do
+          cleanUpDone <- isJust <$> readIORef delayedTestRef
```

Comment:
This test can in theory pass if `check` in line 67 were never run. Can we maybe write something other than `Nothing` (the initial ref state) to be sure the value is coming from deallocation?